packages/components/src/range-control/index.js

```diff
@@ -39,263 +35,257 @@ import {
+	const fillValue = isValueReset ? 50 : calculatedFillValue;
+	const fillValueOffset = `${ clamp( fillValue, 0, 100 ) }%`;
+
+	const classes = classnames( 'components-range-control', className );
+
+	const wrapperClasses = classnames(
+		'components-range-control__wrapper',
+		!! marks && 'is-marked'
+	);
+
+	const id = useInstanceId( RangeControl, 'inspector-range-control' );
+	const describedBy = !! help ? `${ id }__help` : undefined;
+	const enableTooltip = showTooltipProp !== false && isFinite( value );
+
+	const handleOnRangeChange = ( event ) => {
+		const nextValue = parseFloat( event.target.value );
+		handleOnChange( nextValue );
+	};
+
+	const handleOnChange = ( nextValue ) => {
+		if ( isNaN( nextValue ) ) {
+			handleOnReset();
+			return;
+		}
+
+		setValue( nextValue );
+		onChange( nextValue );
+	};
+
+	const handleOnReset = () => {
+		let resetValue = parseFloat( resetFallbackValue );
+		let onChangeResetValue = resetValue;
+
+		if ( isNaN( resetValue ) ) {
+			resetValue = null;
+			onChangeResetValue = undefined;
+		}
+
+		setValue( resetValue );
+
+		/**
+		 * Previously, this callback would always receive undefined as
+		 * an argument. This behavior is unexpected, specifically
+		 * when resetFallbackValue is defined.
+		 *
+		 * The value of undefined is not ideal. Passing it through
+		 * to internal <input /> elements would change it from a
+		 * controlled component to an uncontrolled component.
+		 *
+		 * For now, to minimize unexpected regressions, we're going to
+		 * preserve the undefined callback argument, except when a
+		 * resetFallbackValue is defined.
+		 */
+		onChange( onChangeResetValue );
+	};
+
+	const handleShowTooltip = () => setShowTooltip( true );
+	const handleHideTooltip = () => setShowTooltip( false );
+
+	const handleOnBlur = ( event ) => {
+		onBlur( event );
+		setIsFocused( false );
+		handleHideTooltip();
+	};
+
+	const handleOnFocus = ( event ) => {
+		onFocus( event );
+		setIsFocused( true );
+		handleShowTooltip();
+	};
+
+	const offsetStyle = {
+		[ isRTL ? 'right' : 'left' ]: fillValueOffset,
+	};
+
+	return (
+		<BaseControl
+			className={ classes }
+			label={ label }
+			id={ id }
+			help={ help }
+		>
+			<Root className="components-range-control__root" isRTL={ isRTL }>
+				{ beforeIcon && (
+					<BeforeIconWrapper>
+						<Icon icon={ beforeIcon } />
+					</BeforeIconWrapper>
+				) }
+				<Wrapper
+					className={ wrapperClasses }
+					color={ colorProp }
+					marks={ !! marks }
 				>
-					{ beforeIcon && (
-						<BeforeIconWrapper>
-							<Icon icon={ beforeIcon } />
-						</BeforeIconWrapper>
-					) }
-					<Wrapper
-						className={ wrapperClasses }
-						color={ colorProp }
-						marks={ !! marks }
-					>
-						<InputRange
-							{ ...props }
-							{ ...hoverInteractions }
-							aria-describedby={ describedBy }
-							aria-label={ label }
-							aria-hidden={ false }
-							className="components-range-control__slider"
```

Comment:
I think the new `InputRange` component makes sense. However, I also feel that the `components-range-control__slider` class should be attached here, within the range component, and not in `InputRange` itself. The reason for this is that it's a class dependent on the parent component, and for homogeneity with the rest of child components.

Reply:
I can make that change. For context, the component doesn't use classNames for styling. It maintains this className based selector targeting for consumers (and tests).